reactioncommerce/reaction
# chore: clean-up tasks related to removing `appEvents` from all meteor code

Resolves #5691   
Impact: minor
Type: feature|style|refactor|chore

Remove or refactor all code that calls `appEvents` from meteor code.

### Refactored into mutations
- [`discounts/codes/apply`](https://github.com/reactioncommerce/reaction/blob/develop/imports/plugins/included/discount-codes/server/methods/methods.js#L270)
- [`discounts/codes/remove`](https://github.com/reactioncommerce/reaction/blob/develop/imports/plugins/included/discount-codes/server/methods/methods.js#L140)

### Removed
- [`updateShipmentStatus`](https://github.com/reactioncommerce/reaction/blob/develop/imports/plugins/core/orders/server/util/updateShipmentStatus.js#L53) I don't believe this function is called anywhere in the app - it doesn't even seem to be attached to a meteor method anymore - so we can mostly likely just delete this file
- [`workflow/pushItemWorkflow`](https://github.com/reactioncommerce/reaction/blob/develop/imports/plugins/core/checkout/server/methods/workflow.js#L87) We want to remove `coreItemWorkflow` anyway, maybe this is the time? In addition, this is called from the `checkout` plugin, which I don't believe we don't use anymore, so maybe we remove this plugin?

### TODO
- [`shop/createShop`](https://github.com/reactioncommerce/reaction/blob/develop/imports/plugins/core/core/server/methods/shop/createShop.js#L148) I don't believe this is used anymore, we can most likely just delete this function, along with the Blaze `Template` where it's being called.

## Testing
1. See that the app functions as expected
1. See that discount code applying and removing works with new mutations. We have no UI for this at the moment, so you need to do it by creating a Cart, then taking that cart data and using `GraphiQL` or another GQL interface to perform the action

```
mutation {
  applyDiscountCodeToCart(input: {
    cartId: "ENCODED_CART_ID",
    discountCode: "CODE",
    shopId: "SHOP_ID",
    token: null
  }) {
    clientMutationId
  }
}
```

A discount code is a payment, so the discount code id should be encoded with `reaction/payment`
```
mutation {
  removeDiscountCodeFROMCart(input: {
    cartId: "ENCODED_CART_ID",
    discountCodeID: "ENCODED_PAYMENT_ID",
    shopId: "SHOP_ID",
    token: null
  }) {
    clientMutationId
  }
}
```
